lib/src/common/disposable.dart

```diff
@@ -436,6 +463,7 @@ class Disposable implements _Disposable, DisposableManagerV3, LeakFlagger {
     _internalDisposables.add(disposable);
   }
 
+  @deprecated
```

Comment:
We should discuss whether we actually want to deprecate this or not. Can anyone think of any edge cases where the new method won't work? @Workiva/rich-app-platform-pp